src/main/scala/io/scalac/wtf/domain/User.scala

```diff
@@ -1,11 +1,46 @@
 package io.scalac.wtf.domain
 
+import cats.data.Validated.{invalidNel, valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.implicits._
 import slick.lifted.MappedTo
 
+import scala.util.matching.Regex
 
 case class UserId(value: Long) extends MappedTo[Long]
 
 case class User(
     id: Option[UserId] = None,
     email: String, 
     password: String)
+
+object User {
+
+  sealed trait ValidationError
+  final case object WrongEmailPattern extends ValidationError
+  final case object PasswordTooShort  extends ValidationError
+
+  def validateUser(email: String, password: String): ValidatedNel[ValidationError, User] = {
+    val validation = (validateEmail(email) |@| validatePassword(password))
```

Comment:
Going back to my question that got eaten by code changes - this example doesn't convince me to use Scalaz.
Quite on the contrary - it encourages me to use plain if's (you already use `match`) or something more crazy - exceptions. These are simpler ( |@| yourself :) ), have less boilerplate and don't require any dependencies. 
Validating a two-field structure doesn't really show anything. I would add more fields, so any switch based approach would be out of the question. 
It might be worth to show error stacking (I assume you are doing that already) so throwing Exceptions is invalidated too